app/models/manageiq/providers/microsoft/infra_manager/refresh_parser.rb

```diff
@@ -413,6 +414,17 @@ def process_vm_hardware(vm)
       }
     end
 
+    def process_vm_custom_properties(vm)
+      vm['CustomProperty'].map do |key, value|
```

Comment:
Can you use `vm['CustomProperty']&.map` to handle the case where this is nil?  That will also fix the spec test failures